docs/content/integrations/fivetran/fivetran.mdx

```diff
@@ -0,0 +1,195 @@
+
+```python file=/integrations/fivetran/representing-fivetran-assets.py
+from dagster_fivetran import FivetranWorkspace, load_fivetran_asset_specs
+
+import dagster as dg
+
+fivetran_workspace = FivetranWorkspace(
+    account_id=dg.EnvVar("FIVETRAN_ACCOUNT_ID"),
+    api_key=dg.EnvVar("FIVETRAN_API_KEY"),
+    api_secret=dg.EnvVar("FIVETRAN_API_SECRET"),
+)
+
+fivetran_specs = load_fivetran_asset_specs(fivetran_workspace)
+defs = dg.Definitions(
+    assets=[*fivetran_specs], resources={"fivetran": fivetran_workspace}
+)
+```
+
+### Sync and materialize Fivetran assets
+
+You can use Dagster to sync Fivetran connectors and materialize Fivetran connector tables. You can use the <PyObject module="dagster_fivetran" method="fivetran_assets" /> decorator to create the assets definition for a given connector, or the <PyObject module="dagster_fivetran" method="build_fivetran_assets_definitions" /> factory to create all assets definitions for your Fivetran workspace.
+
+```python file=/integrations/fivetran/sync-and-materialize-fivetran-assets.py
+from dagster_fivetran import (
+    FivetranWorkspace,
+    build_fivetran_assets_definitions,
+    fivetran_assets,
+)
+
+import dagster as dg
+
+fivetran_workspace = FivetranWorkspace(
+    account_id=dg.EnvVar("FIVETRAN_ACCOUNT_ID"),
+    api_key=dg.EnvVar("FIVETRAN_API_KEY"),
+    api_secret=dg.EnvVar("FIVETRAN_API_SECRET"),
+)
+
+
+# Creating assets definition for a given connector using the `@fivetran_assets` decorator
+@fivetran_assets(
+    connector_id="fivetran_connector_id",
+    name="fivetran_connector_id",
+    group_name="fivetran_connector_id",
+    workspace=fivetran_workspace,
+)
+def fivetran_connector_assets(
+    context: dg.AssetExecutionContext, fivetran: FivetranWorkspace
+):
+    yield from fivetran.sync_and_poll(context=context)
+
+
+# Alternatively, creating all assets definitions for the Fivetran workspace
+# using the `build_fivetran_assets_definitions` factory
+all_fivetran_assets = build_fivetran_assets_definitions(fivetran_workspace)
+
+defs = dg.Definitions(
+    assets=[*all_fivetran_assets],
+    resources={"fivetran": fivetran_workspace},
+)
+```
+
+### Customize asset definition metadata for Fivetran assets
+
+By default, Dagster will generate asset specs for each Fivetran asset and populate default metadata. You can further customize asset properties by passing a custom <PyObject module="dagster_fivetran" object="DagsterFivetranTranslator" /> subclass to the <PyObject module="dagster_fivetran" method="load_fivetran_asset_specs" /> function, the <PyObject module="dagster_fivetran" method="fivetran_assets" /> decorator or the <PyObject module="dagster_fivetran" method="build_fivetran_assets_definitions" /> factory.
+
+```python file=/integrations/fivetran/customize-fivetran-asset-defs.py
```

Comment:
This whole code snippet feels really long, can we make use of `startbefore` and `endafter` to trim this down to only the relevant bits?

Reply:
I updated it in [0d36d97](https://github.com/dagster-io/dagster/pull/26026/commits/0d36d97fc555ba84a75e34bb0cf899eb1a49d30a). The code snippet only showcase the load_fivetran_asset_specs use case now, and I added a note after the example saying that passing a custom translator can also be done with the decorator and the factory.